### ISSUE TITLE
Make file system tree creation more efficient.

### ISSUE THREAD

**Issue by goggle** [goggle/flatten#3]
In the file `filesystem/tree.go` there is a function `create`, which creates a file system tree in memory. At the moment it is completely inefficient. This code should be optimized.

**Comment by andrefreitas** [comment]
Hi, I can take a look at this. 

By looking in the `create` method code, there are unnecessary iterations when a file doesn't have a parent already inserted in the tree. So it keeps retrying inserting these files, until their parents are inserted.

**Comment by goggle** [comment]
Yes, that's exactly the problem. Maybe we can add an option to the `create` function, that automatically creates the underlying directories, if they do not exist yet.

**Comment by andrefreitas** [comment]
If the cost of sorting directories and files entries by the number of "/" before inserting them in the tree would be considered acceptable, could be an option.
